src/GraphQL/Utilities/LinkConfiguration.cs

```diff
@@ -0,0 +1,318 @@
+            return value;
+        if (Namespace == null)
+            throw new InvalidOperationException("The specified type has not been imported.");
+        return Namespace + "__" + typeName;
+    }
+
+    /// <summary>
+    /// Returns the aliased name for the requested directive.
+    /// For directives that have been explicitly imported, it returns the configured alias, or when no alias is configured, the directive name.
+    /// For other directives, it returns the directive name prefixed with the namespace.
+    /// </summary>
+    /// <remarks>
+    /// Does not prefix the returned directive name with '@'.
+    /// </remarks>
+    /// <param name="directiveName">The name of the directive to get the name for, without the '@' prefix.</param>
+    public string NameForDirective(string directiveName)
+    {
+        var withAt = "@" + directiveName;
+        if (Imports.TryGetValue(withAt, out var value))
+            return value == withAt ? directiveName : value.Substring(1);
+        if (Namespace == null)
+            throw new InvalidOperationException("The specified type has not been imported.");
+        return Namespace + "__" + directiveName;
+    }
+
+    /// <summary>
+    /// Configures the specified applied directive with the settings from this configuration.
+    /// </summary>
+    /// <remarks>
+    /// This is called by <see cref="SchemaExtensions.LinkSchema(ISchema, string, Action{LinkConfiguration}?)"/>
+    /// and <see cref="SchemaExtensions.AddLinkDirectiveSupport(ISchema, Action{LinkConfiguration}?)"/>
+    /// to apply the configuration to the schema.
+    /// </remarks>
+    internal void ConfigureAppliedDirective(AppliedDirective o)
+    {
+        if (o.Name != "link")
+            throw new ArgumentException("The supplied directive is not a @link directive.", nameof(o));
+
+        o.WithMetadata(METADATA_KEY, this);
+
+        o.List?.Clear();
+
+        o.AddArgument(new DirectiveArgument("url") { Value = Url });
+
+        if (Namespace != null && Namespace != _originalNamespace)
+        {
+            o.AddArgument(new DirectiveArgument("as") { Value = Namespace });
+        }
+
+        if (Purpose != null)
+            o.AddArgument(new DirectiveArgument("purpose") { Value = Purpose.Value });
+
+        if (Imports.Count > 0)
+        {
+            var importList = new List<object>();
+            foreach (var import in Imports)
+            {
+                if (import.Key == null)
+                    throw new InvalidOperationException("No name specified for an import.");
+                var alias = import.Value ?? import.Key;
+                if (import.Key.StartsWith("@", StringComparison.Ordinal))
+                {
+                    NameValidator.ValidateName(import.Key.Substring(1), NamedElement.Directive);
+                    if (!alias.StartsWith("@", StringComparison.Ordinal))
+                        throw new InvalidOperationException("An alias for a directive must start with '@'.");
```

Comment:
Please specify the directive name in the error message

Reply:
Updated